internal/client/job.go

```diff
@@ -235,18 +237,45 @@ func (c *Project) doJobMonitored(ctx context.Context, job *pb.Job, ui terminal.U
 		}
 	} else {
 		var configRunner *configpkg.Runner
-		// Find runner configuration on the app
+		// Find runner config on the app
 		if job.Application != nil {
 			configRunner = c.waypointHCL.ConfigAppRunner(job.Application.Application)
 		}
-		// If not on app, try to find it on the project
-		if configRunner == nil {
+		// Find runner config on the project
+		if configRunner != nil {
+			if c.waypointHCL.ConfigRunner() != nil {
+				log.Warn("runner configurations found on both app and project, app takes precedence.")
+			}
+		} else {
 			configRunner = c.waypointHCL.ConfigRunner()
 		}
-		// If runner config is found, assign to job
+		// If runner config exists, assign to job
 		if configRunner != nil {
-			job.OndemandRunner = &pb.Ref_OnDemandRunnerConfig{
-				Name: configRunner.Profile,
+			if configRunner.Profile != "" {
+				job.OndemandRunner = &pb.Ref_OnDemandRunnerConfig{
+					Name: configRunner.Profile,
+				}
+			}
+			if configRunner.TargetLabels != nil {
+				if job.OndemandRunner != nil {
+					log.Warn("both runner profile and target labels are set, profile takes precedence.")
+				} else {
+					runners, err := c.client.ListRunners(ctx, &pb.ListRunnersRequest{})
+					if err != nil {
+						errors.Wrapf(err, "no runners found.")
```

Comment:
Since we're using the `status`/`codes` errors else where in this file, this would probably be better to return a `status.NotFound` error

Reply:
We could also just return the error the `ListRunners` endpoint returns, although if we want to wrap it, we could mention the job client failed to list runners or something more specific to this file?

Reply:
^ realized that it would not error and simply return an empty list, so modified this code behavior.